File: coding-test/java-kotlin/src/kr/co/programmers/java/greedy/LifeBoat.java
package kr.co.programmers.java.greedy;

import java.util.Arrays;

public class LifeBoat {

    public static void main(String[] args) {
        int[] people = {70, 50, 80, 50};
        int limit = 100;
        System.out.println(new Solution().solution(people, limit));
    }

    static class Solution {
        public int solution(int[] people, int limit) {
            Arrays.sort(people);
            int i = 0, j = people.length - 1;
            for (; i < j; --j) {
                if (people[i] + people[j] <= limit)
                    ++i;
            }
            return people.length - i;
        }
    }
}

File: practice-coding-test/javascript/src/algorithm/greedy/life-boat.js
// @ts-check

/**
 * @param {number[]} people
 * @param {number} limit
 * @returns {number}
 */
function solution(people, limit) {
  people = people.sort((a, b) => b - a)
  let start = 0,
    end = people.length - 1
  for (; start < end; start += 1) {
    if (people[start] + people[end] <= 100) end -= 1
  }
  return start
}

const people = [70, 50, 80, 50]
const limit = 100
console.log(solution(people, limit))
